### ISSUE TITLE
Can't add custom IIS step to project with deployment type Web Application or Virtual Directory

### ISSUE THREAD

**Issue by KennethBates** [OctopusDeploy/Issues#6663]
# Prerequisites

- [x] I have verified the problem exists in the latest version
- [x] I have searched [open](https://github.com/OctopusDeploy/Issues/issues) and [closed](https://github.com/OctopusDeploy/Issues/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aclosed) issues to make sure it isn't already reported
- [x] I have written a descriptive issue title
- [x] I have linked the original source of this report
- [x] I have tagged the issue appropriately (area/*, kind/bug, tag/regression?)

# The bug

If you configure a custom library step template based on the built-in Deploy to IIS step, and set the `Deployment Type` to anything other than `IIS Website` (i.e. `IIS Virtual Directory` or `IIS Web Application`), this step fails to be added to a project with validation errors `Please enter an IIS web site name` and `Please enter an IIS application pool name`.

## What I expected to happen

It would pass validation, as it does when configuring the step directly in a project's deployment process, where the step is added without this error.

## Steps to reproduce

1. Create a custom step template in the library based off the built-in `Deploy to IIS` step template
2. Configure the `Deployment Type` to `IIS Web Application` or `IIS Virtual Directory`
3. Attempt to add this template to a project

### Screen capture

![custom_iis_step_validation_error](https://user-images.githubusercontent.com/20388190/98614951-427aed00-2345-11eb-8c92-c5b4c5d76ad8.png)

### Log exerpt

```
Please enter an IIS web site name
Please enter an IIS application pool name
```

## Affected versions

Reported in 2020.4.6 & 2020.4.11 and reproduced in 2020.5.0

## Workarounds

In the custom step template, change the `Deployment Type` to `IIS Website`, set the fields (site name and app pool name as mentioned in the error) and save. Then change it back to the desired type and save again.

## Links

Report: https://octopus.zendesk.com/agent/tickets/41917 (private)

**Comment by adrianhowchin** [comment]
We have flagged this as an issue with product, but at this point we do not have capacity to assign an engineer to resolve the issue. We'll keep the issue open, but unfortunately we can not provide any guidance as to when it will be resolved.

**Comment by N-lson** [comment]
I'm unable to reproduce this issue on `release/2020.6` and `master` branches. I've tested using the provided step template in Zendesk and also by creating a new step template. @KennethBates could you confirm if you're able to reproduce this issue on either of these branches?

**Comment by droyad** [comment]
I've reproduced this on `2020.5` but can't on `2020.6`. I suspect it was fixed as part of  #6648
